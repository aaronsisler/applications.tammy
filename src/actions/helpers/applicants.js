export const setApplicants = (applicants) => ({
    type: 'SET_APPLICANTS',
    applicants
});
