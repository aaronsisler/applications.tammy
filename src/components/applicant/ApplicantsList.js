import React from 'react';
import PropTypes from 'prop-types';
import ApplicantsListItem from 'Applicant/ApplicantsListItem';

export default class ApplicantsList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.applicants.length === 0) {
            return (
                <div className="applicants_list empty">
                    No available items
                </div>
            );
        }

        return (
            <div className="applicants_list">
                {
                    this.props.applicants.map((applicant) =>
                        <ApplicantsListItem
                            key={applicant.applicantId}
                            applicantId={applicant.applicantId}
                            displayName={applicant.user.displayName}
                            firstName={applicant.user.firstName}
                            lastName={applicant.user.lastName}
                            positionId={this.props.positionId}
                        />)
                }
            </div>
        );
    }
}

ApplicantsList.propTypes = {
    applicants: PropTypes.array.isRequired,
    positionId: PropTypes.string.isRequired,
};
