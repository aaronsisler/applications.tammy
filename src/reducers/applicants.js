const applicantsReducerDefaultState = [];

const applicantsReducer = (state = applicantsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS':
            return action.applicants || applicantsReducerDefaultState;
        default:
            return state;
    }
};

export default applicantsReducer;

