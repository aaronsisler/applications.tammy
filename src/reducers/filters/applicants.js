const applicantsFilterReducerDefaultState =
{
    text: '',
};

const applicantsFilterReducer = (state = applicantsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        default:
            return state;
    }
};

export default applicantsFilterReducer;
