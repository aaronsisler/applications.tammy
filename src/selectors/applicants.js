// Get filtered applicants
export default (applicants = [], { text = '' }) => applicants
    .filter((applicant) => {
        const lastNameMatch = applicant.user.lastName.toLowerCase().includes(text.toLowerCase());
        const firstNameMatch = applicant.user.firstName.toLowerCase().includes(text.toLowerCase());
        return lastNameMatch || firstNameMatch;
    })
