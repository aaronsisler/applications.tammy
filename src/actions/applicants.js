import database from 'Firebase/firebase';
import { setApplicants } from 'Actions/helpers/applicants';
import { setApplicantsTextFilter } from 'Actions/filters/applicantsText';

export const startSetApplicants = () => (dispatch, getState) => {
    const { positionId } = getState().workflow.position;

    return database.ref(`applicants/${positionId}`).once('value').then((snapshot) => {
        const applicants = [];
        snapshot.forEach((childSnapshot) => {
            const applicant = childSnapshot.val();
            const applicantNotes = applicant.applicantNotes
                ? [...Object.values(applicant.applicantNotes)]
                : [];
            applicants.push({
                applicantId: childSnapshot.key,
                positionId,
                ...childSnapshot.val(),
                applicantNotes,
            });
        });

        return dispatch(setApplicants(applicants));
    });
}

export const startSetApplicantsTextFilter = (text) => (dispatch) => dispatch(setApplicantsTextFilter(text));
