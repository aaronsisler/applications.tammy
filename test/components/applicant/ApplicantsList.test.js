import React from 'react';
import { shallow } from 'enzyme';
import ApplicantsList from 'Applicant/ApplicantsList';
import applicants from '../../fixtures/applicants';

describe('ApplicantsList', () => {
    let wrapper;

    const buildWrapper = (applicantsInput = []) => {
        wrapper = shallow(
            <ApplicantsList
                applicants={applicantsInput}
            />);
    }

    it('should render ApplicantsList correctly when applicants are available', () => {
        buildWrapper(applicants);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render ApplicantsList correctly when no applicants are available', () => {
        buildWrapper();

        expect(wrapper).toMatchSnapshot();
    });
});
