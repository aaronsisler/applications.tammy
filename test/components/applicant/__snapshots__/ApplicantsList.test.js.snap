// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicantsList should render ApplicantsList correctly when applicants are available 1`] = `
<div
  className="applicants_list"
>
  <Connect(ApplicantsListItem)
    applicantId="firstApplicantId"
    key="firstApplicantId"
  />
  <Connect(ApplicantsListItem)
    applicantId="secondApplicantId"
    key="secondApplicantId"
  />
</div>
`;

exports[`ApplicantsList should render ApplicantsList correctly when no applicants are available 1`] = `
<div
  className="applicants_list empty"
>
  No available items
</div>
`;
